formats/cbor/commonTest/src/kotlinx/serialization/cbor/CborReaderTest.kt

```diff
@@ -104,4 +108,338 @@ class CborReaderTest {
             )
         }
     }
+
+    @Test
+    fun testIgnoreUnknownKeysFailsWhenKotlinClassIsMissingKeys() {
+        // with maps & lists of indefinite length
+        assertEquals(
+            expected = "Field 'a' is required, but it was missing",
+            actual = assertFailsWith<SerializationException> {
+                lenient.decodeFromHexString(
+                    Simple.serializer(),
+                    "bf637374726d48656c6c6f2c20776f726c64216169182a686e756c6c61626c65f6646c6973749f61616162ff636d6170bf01f502f4ff65696e6e6572bf6161636c6f6cff6a696e6e6572734c6973749fbf6161636b656bffffff"
+                )
+            }.message
+        )
+
+        // with maps & lists of definite length
+        assertEquals(
+            expected = "Field 'a' is required, but it was missing",
+            actual = assertFailsWith<SerializationException> {
+                lenient.decodeFromHexString(
+                    Simple.serializer(),
+                    "a7646c6973748261616162686e756c6c61626c65f6636d6170a202f401f56169182a6a696e6e6572734c69737481a16161636b656b637374726d48656c6c6f2c20776f726c642165696e6e6572a16161636c6f6c"
+                )
+            }.message
+        )
+    }
+
+    /**
+     * Tests skipping unknown keys associated with values of the following CBOR types:
+     * - Major type 0: an unsigned integer
+     * - Major type 1: a negative integer
+     * - Major type 2: a byte string
+     * - Major type 3: a text string
+     */
+    @Test
+    fun testSkipPrimitives() {
+        /* A5                           # map(5)
+         *    61                        # text(1)
+         *       61                     # "a"
+         *    01                        # unsigned(1)
+         *    61                        # text(1)
+         *       62                     # "b"
+         *    20                        # negative(0)
+         *    61                        # text(1)
+         *       63                     # "c"
+         *    42                        # bytes(2)
+         *       CAFE                   # "\xCA\xFE"
+         *    61                        # text(1)
+         *       64                     # "d"
+         *    6B                        # text(11)
+         *       48656C6C6F20776F726C64 # "Hello world"
+         *    61                        # text(1)
+         *       65                     # "e"
+         *    02                        # unsigned(2)
+         */
```

Comment:
Are comments describing the CBOR encoded data helpful or just noise?

_A developer who later encounters this code could themselves paste the `withDecoder` `input` into [cbor.me](http://cbor.me/), so having it as a comment feels somewhat superfluous, although it can be rather convenient having it quickly available for reference. Downside being that it's a comment that needs to be maintained if any changes to `input` are made._

Reply:
I think they are more helpful than harmful. Again, test data changes relatively rarely.